src/app/work-item/work-item.service.ts

```diff
@@ -435,16 +436,75 @@ export class WorkItemService {
    * place should be implemented
    */
   moveItem(wi: WorkItem, dir: String) {
-    //We need to call an ordering API which will store
-    //the new location for the selected work item
     let index = this.workItems.findIndex(x => x.id == wi.id);
-    let wiSplice = this.workItems.splice(index, 1);
-    if (dir == 'top') {
-      this.workItems.splice(0, 0, wi);
-    } else {
-      this.workItems.splice(this.workItems.length, 0, wi);
+    wi.attributes.nextitem = '';
+    wi.attributes.previousitem = '';
+    switch (dir){
+      case 'top':
+        //this.workItems.splice(0, 0, wi);
+        break;
+      case 'bottom':
+        //this.workItems.splice(this.workItems.length, 0, wi);
+        break;
+      case 'up':
+        if (index > 0) { //no moving of element if it is the first element
+          //move the work item up by 1. Below statement will create two elements
+          this.workItems.splice( index - 1 , 0, wi);
+          //remove the duplicate element
+          this.workItems.splice( index + 1, 1);
+          //Set the previous and next WI ids
+          if (index === 1) {
+            wi.attributes.nextitem = parseInt(this.workItems[index].id);
```

Comment:
This line and 460 are same and looks repetation. So, you can replace following code with 456 to 461.
```
wi.attributes.nextitem = parseInt(this.workItems[index].id);
if (index !== 1) {
    wi.attributes.previousitem = parseInt(this.workItems[index - 2].id);
}
```